src/new/wallet.js

```diff
@@ -531,6 +538,131 @@ class HathorWallet extends EventEmitter {
     return balance;
   }
 
+  /**
+   * Create a new token for this wallet
+   *
+   * @param {String} name Name of the token
+   * @param {String} symbol Symbol of the token
+   * @param {number} amount Quantity of the token to be minted
+   * @param {String} address Optional parameter for the destination of the created token
+   *
+   * @return {Object} Object with {success: true, sendTransaction, promise}, where sendTransaction is a
+   * SendTransaction object that emit events while the tx is being sent and promise resolves when the sending is done
+   **/
+  createNewToken(name, symbol, amount, address) {
+    storage.setStore(this.store);
+    const mintAddress = address || this.getCurrentAddress();
+    const ret = tokens.createToken(mintAddress, name, symbol, amount, this.pinCode);
+
+    if (ret.success) {
+      const sendTransaction = ret.sendTransaction;
+      sendTransaction.start();
+      return {success: true, promise: sendTransaction.promise, sendTransaction};
+    } else {
+      return ret;
+    }
+
+  }
+
+  /**
+   * Select authority utxo for mint or melt. Depends on the callback received as parameter
+   *
+   * @param {String} tokenUid UID of the token to select the authority utxo
+   * @param {function} isUtxoCallback Callback to check if the output is the authority I wanet (isMeltOutput or isMintOutput)
+   *
+   * @return {Object} Object with {tx_id, index, address} of the authority output. Returns null in case there are no utxos for this type
+   **/
+  selectAuthorityUtxo(tokenUid, isUtxoCallback) {
```

Comment:
I'd name it `filterUTXOs()`. We could add an `{options}` parameter to allow common filters (e.g. mint authority, melt authority, p2pkh).

Reply:
Do you think we should add this `options` parameter right now? I see it as adding more confusion to the code right now (we would need a map from the option to the callback method).

Anyway, I agree with you that the name `filterUTXOs` is better than `isUtxoCallback`, just want to make sure if I should go with the callback method or the `options` parameter? In my opinion we should keep only the callback, what do you think?

Reply:
We can keep the callback and refactor it later to include more options. Just leave a comment so we know how to improve it later.

Reply:
Done!